GitUI/Plugin/PluginRegistry.cs

```diff
@@ -13,26 +15,44 @@ public static class PluginRegistry
 
         public static List<IRepositoryHostPlugin> GitHosters { get; } = new List<IRepositoryHostPlugin>();
 
-        public static void Initialize()
+        public static bool PluginsRegistered { get; private set; }
+
+        /// <summary>
+        /// Initialises all available plugins on the background thread.
+        /// </summary>
+        /// <param name="postInitialiseAsync">A function to execute once plugins are loaded.</param>
+        public static Task InitializeAsync(Func<Task> postInitialiseAsync)
```

Comment:
💡  Passing continuations via a callback argument reminds me of the days before async/await. It would be better to reorganize this code to avoid the need for this form.